File: services/mobile/lib/src/lyrics/lyric_line.dart
/// A single word-level timing fragment within a [LyricLine], used for
/// karaoke-style progressive highlighting.
class LyricWord {
  const LyricWord({required this.offset, required this.text});
  final Duration offset;
  final String text;
}

/// A single line of lyrics with its start timestamp.
class LyricLine {
  const LyricLine({
    required this.timestamp,
    required this.text,
    this.words,
    this.translation,
  });
  final Duration timestamp;
  final String text;
  /// Word-level timing fragments for karaoke highlighting, or null/empty
  /// when the source line has no inline `<mm:ss.xx>` tags.
  final List<LyricWord>? words;
  /// Aligned translation text for this line, or null when no translation
  /// was uploaded or the translation line count didn't match.
  final String? translation;
}

File: services/web/lib/lyrics/lyricLine.ts
/** A single word-level timing fragment within a LyricLine, used for karaoke-style progressive highlighting. */
export interface LyricWord {
  /** Offset in milliseconds from the start of the track. */
  offsetMs: number;
  text: string;
}

/** A single line of lyrics with its start timestamp. */
export interface LyricLine {
  /** Timestamp in milliseconds from the start of the track. */
  timestampMs: number;
  text: string;
  /** Word-level timing fragments for karaoke highlighting, or undefined when the source line has no inline `<mm:ss.xx>` tags. */
  words?: LyricWord[];
  /** Aligned translation text for this line, or undefined when no translation was uploaded or the translation line count didn't match. */
  translation?: string;
}
